tutorials/unit_tutorial.py

```diff
@@ -0,0 +1,128 @@
+you on your way to utilizing the computation functions in MetPy.
+"""
+
+import numpy as np
+
+from metpy.units import units
+
+#########################################################################
+# Simple Calculation
+# ------------------
+#
+# Let's say we want to calculate the area of a rectangle. It so happens that
+# one of our colleagues measures their side of the rectangle in imperial units
+# and the other in metric units. No problem! First we need to attach units to
+# our measurements. For many units the easiest way is by find the unit as an
+# attribute of the unit registry:
+
+length = 10.4 * units.inches
+width = 20 * units.meters
+print(length, width)
+
+#########################################################################
+# Don't forget that you can use tab completion to see what units are available!
+# Just about every imaginable quantity is there, but if you find one that isn't,
+# we're happy to talk about adding it.
+#
+# While it may seem like a lot of trouble, let's compute the area of a rectangle
+# defined by our length and width variables above. Without units attached, you'd
+# need to remember to perform a unit conversion before multiplying or you would
+# end up with an area in inch-meters and likely forget about it. With units
+# attached, the units are tracked for you.
+
+area = length * width
+print(area)
+
+#########################################################################
+# That's great, now we have an area, but it is not in a very useful unit still.
+# Units can be converted using the `to()` method. While you won't see square meters in
+# the units list, we can parse complex/compound units as strings:
+
+print(area.to('m^2'))
+
+#########################################################################
+# Temperature
+# -----------
+# Temperature units are actually relatively tricky (more like absolutely tricky as
+# you'll see). Temperature is a non-multiplicative unit - they are in a system
+# with a reference point. That means that not only is there a scaling factor, but
+# also an offset. This makes the math and unit book-keeping a little more complex.
+# Imagine adding 10 degrees Celsius to 100 degrees Celsius. Is the answer 110
+# degrees Celsius or 383.15 degrees Celsius (283.15 K + 272.15 K)? That's why
+# there are delta degrees units in the unit registry for offset units. For more
+# examples and explanation you can watch MetPy Monday #13:
+# https://www.youtube.com/watch?v=iveJCqxe3Z4.
+#
+# Let's take a look at how this works and fails:
+#
+# We would expect this to fail because we cannot add two offset units (and it does
+# fail as an "Ambiguous operation with offset unit").
+#
+# `10 * units.degC + 5 * units.degC`
+#
+# On the other hand, we can subtract two offset
+# quantities and get a delta:
+
+print(10 * units.degC - 5 * units.degC)
+
+#########################################################################
+# We can add a delta to an offset unit as well:
```

Comment:
I think it would be good to have a bit of prose explaining delta vs. regular unit